src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -344,9 +344,23 @@ private static byte[] GetSqlServerSPN(DataSource dataSource)
         private static byte[] GetSqlServerSPN(string hostNameOrAddress, string portOrInstanceName)
         {
             Debug.Assert(!string.IsNullOrWhiteSpace(hostNameOrAddress));
-            IPHostEntry hostEntry = Dns.GetHostEntry(hostNameOrAddress);
-            string fullyQualifiedDomainName = hostEntry.HostName;
-            string serverSpn = SqlServerSpnHeader + "/" + fullyQualifiedDomainName;
+            IPHostEntry hostEntry = null;
+            string fullyQualifiedDomainName;
+            try
+            {
+                hostEntry = Dns.GetHostEntry(hostNameOrAddress);
+            }
+            catch
+            {
+                // An exception can occur while resolving the hostname. We dont care about this exception.
```

Comment:
Nit: dont => don't

Reply:
Are there specific exception types we want to catch, or we really just want to eat everything?

Reply:
Ideally I would like to catch specific exceptions. However the docs at https://msdn.microsoft.com/en-us/library/ms143998.aspx don't mention what will be thrown in what scenario.
Hence the decision to catch everything and fallback on the user provided SPN.

Reply:
Digging through docs.microsoft.com I see that there are specific exceptions that can be caught. https://docs.microsoft.com/en-us/dotnet/api/system.net.dns.gethostentry?view=netframework-4.7.1